src/main/kotlin/app/revanced/patches/all/telephony/sim/spoof/SpoofSimCountryPatch.kt

```diff
@@ -0,0 +1,76 @@
+import app.revanced.patcher.extensions.InstructionExtensions.removeInstruction
+import app.revanced.patcher.extensions.InstructionExtensions.replaceInstruction
+import app.revanced.patcher.patch.annotation.Patch
+import app.revanced.patcher.patch.options.types.StringPatchOption.Companion.stringPatchOption
+import app.revanced.patcher.util.proxy.mutableTypes.MutableMethod
+import app.revanced.util.patch.AbstractTransformInstructionsPatch
+import com.android.tools.smali.dexlib2.builder.instruction.BuilderInstruction11x
+import com.android.tools.smali.dexlib2.iface.ClassDef
+import com.android.tools.smali.dexlib2.iface.Method
+import com.android.tools.smali.dexlib2.iface.instruction.Instruction
+import com.android.tools.smali.dexlib2.iface.instruction.ReferenceInstruction
+import com.android.tools.smali.dexlib2.iface.reference.MethodReference
+import java.util.*
+
+
+@Patch(
+    name = "Spoof SIM country",
+    description = "Spoofs country information returned by the SIM card provider.",
+    use = false,
+)
+@Suppress("unused")
+object SpoofSimCountryPatch : AbstractTransformInstructionsPatch<Pair<Int, String>>() {
+    private val networkCountryIso by stringPatchOption(
+        "getNetworkCountryIso",
+        null,
+        "Network ISO Country Code",
+        "ISO-3166-1 alpha-2 country code equivalent of the MCC (Mobile Country Code) " +
+                "of the current registered operator or the cell nearby.",
+        validator = { it?.uppercase() in Locale.getISOCountries() || it == null }
+    )
+
+    private val simCountryIso by stringPatchOption(
+        "getSimCountryIso",
+        null,
+        "Sim ISO Country Code",
+        "ISO-3166-1 alpha-2 country code equivalent for the SIM provider's country code.",
+        validator = { it?.uppercase() in Locale.getISOCountries() || it == null }
+    )
+
+    override fun filterMap(
+        classDef: ClassDef,
+        method: Method,
+        instruction: Instruction,
+        instructionIndex: Int
+    ): Pair<Int, String>? {
+        if (instruction !is ReferenceInstruction) return null
+
+        val reference = instruction.reference as? MethodReference ?: return null
+        if (reference.definingClass != "Landroid/telephony/TelephonyManager;") return null
+
+        if (!options.contains(reference.name)) return null
+        return (options[reference.name].value as? String)?.let { instructionIndex to it.trim() }
+    }
+
+    override fun transform(
+        mutableMethod: MutableMethod,
+        entry: Pair<Int, String>
+    ) = transformMethodCall(entry, mutableMethod)
+
+    private fun transformMethodCall(
+        entry: Pair<Int, String>,
+        mutableMethod: MutableMethod
+    ) {
+        val (instructionIndex, methodCallValue) = entry
+
+        // Get register of the next MOVE_RESULT_OBJECT instruction that accompanies INVOKE_VIRTUAL.
+        val instruction = mutableMethod.getInstruction<BuilderInstruction11x>(instructionIndex + 1)
```

Comment:
MOVE_RESULT_OBJECT is a OneRegisterInstruction, you should use that as the generic type parameter.